examples/hex-game/src/lib.rs

```diff
@@ -0,0 +1,247 @@
+    pub fn new(size: u16) -> Self {
+        let size_usize = usize::from(size);
+        let cell_count = size_usize
+            .checked_mul(size_usize)
+            .expect("Board is too large.");
+        let cells = vec![Cell::default(); cell_count];
+        Board {
+            size,
+            cells,
+            active: Player::One,
+        }
+    }
+
+    /// Updates the board: The active player places a stone on cell `(x, y)`.
+    ///
+    /// Panics if the move is invalid.
+    pub fn make_move(&mut self, x: u16, y: u16) -> MoveOutcome {
+        assert!(self.winner().is_none(), "Game has ended.");
+        assert!(x < self.size && y < self.size, "Invalid coordinates.");
+        assert!(self.cell(x, y).stone.is_none(), "Cell is not empty.");
+        self.place_stone(x, y);
+        if let Some(winner) = self.winner() {
+            return MoveOutcome::Winner(winner);
+        }
+        MoveOutcome::Ok
+    }
+
+    /// Places the active player's stone on the given cell, updates all cells'
+    /// `connected` flags accordingly. The new cell _must_ be empty!
+    fn place_stone(&mut self, x: u16, y: u16) {
+        let player = self.active;
+        self.cell_mut(x, y).stone = Some(player);
+        self.active = player.other();
+        if !((x == 0 && player == Player::One)
+            || (y == 0 && player == Player::Two)
+            || self
+                .neighbors(x, y)
+                .any(|(nx, ny)| self.cell(nx, ny).is_connected(player)))
+        {
+            return;
+        }
+        let mut stack = vec![(x, y)];
+        while let Some((x, y)) = stack.pop() {
+            self.cell_mut(x, y).connected = true;
+            stack.extend(self.neighbors(x, y).filter(|(nx, ny)| {
+                let cell = self.cell(*nx, *ny);
+                !cell.connected && cell.stone == Some(player)
+            }));
+        }
+    }
+
+    /// Returns the winner, or `None` if the game is still in progress.
+    pub fn winner(&self) -> Option<Player> {
+        let s = self.size - 1;
+        for i in 0..self.size {
+            if self.cell(s, i).is_connected(Player::One) {
+                return Some(Player::One);
+            }
+            if self.cell(i, s).is_connected(Player::Two) {
+                return Some(Player::Two);
+            }
+        }
+        None
+    }
+
+    /// Returns the `Owner` controlling the active player.
+    pub fn active_player(&self) -> Player {
+        self.active
+    }
+
+    /// Returns the cell `(x, y)`.
+    fn cell(&self, x: u16, y: u16) -> Cell {
+        self.cells[x as usize + (y as usize) * (self.size as usize)]
+    }
+
+    /// Returns a mutable reference to cell `(x, y)`.
+    fn cell_mut(&mut self, x: u16, y: u16) -> &mut Cell {
+        &mut self.cells[x as usize + (y as usize) * (self.size as usize)]
+    }
+
+    /// Returns all neighbors of cell `(x, y)`, in the order:
+    /// left, top left, right, bottom right, top right, bottom left
```

Comment:
Is there a reason for this order?

Reply:
It's kind of ordered by the constraints under which that neighbor exists: Is `x == 0`? Is `y == 0` etc. It doesn't matter for the game.